src/lesson2-homework.ts

```diff
@@ -1,109 +1,218 @@
 }
 // TODO: compile and run the code
 excercise4();
 
 // Create a function which uses type alias to calculate the distance between two points in 2D space - points are objects with x and y properties
-function excercise5() {
+function excercise5(): void {
   // TODO: declare a type alias for a point in 2D space (TPoint) - object with x and y properties
   // TODO: declare two variables of type TPoint
   // TODO: assign values to the variables (1,1) and (4,5)
   // TODO: create a function which calculates the distance between two points in 2D space
-  type TPoint = { /* replace  with your code */ x: "" };
+  type TPoint = { x: number, y: number };
+  const point1: TPoint = {x:10, y: 1}
+  const point2: TPoint = {x:4, y: 5}
   function distance(p1: TPoint, p2: TPoint): number {
-    const x1 = 0; // TODO: replace with the first element of p1
-    const y1 = 0; // TODO: replace with the second element of p1
-    const x2 = 0; // TODO: replace with the first element of p2
-    const y2 = 0; // TODO: replace with the second element of p2
-    // TODO: use distructuring to get x and y from p1 and p2
+    const {x:x1, y:y1} = p1
+    const {x:x2, y:y2} = p2
     // TODO: calculate the distance
-    return 0;
+    return Math.sqrt((x2 - x1)**2 + (y2 - y1)**2);
   }
-  // TODO: call the function and print the result to console
+  console.log(`Exercise 5. Distance is: ${distance(point1, point2)}`)
 }
 // TODO: compile and run the code
 excercise5();
 
 // Create functions that use const declarations
-function excercise6() {
+function excercise6(): void {
   // TODO: declare a const PI and assign value 3.14
   // TODO: declare a function which calculates a circle area, takes radius as a parameter
   // TODO: call the function and print the result to console
   // TODO: check the type of PI variable
   // TODO: declare a const variable that is an object with two properties - name and age
   // TODO: declare a function which takes a person object as a parameter and increments age by 1
   // TODO: call the function and print the person object to console
+  const PI = 3.14
+  const circleArea = (radius: number): number => {
+    return PI*(radius**2)
+  }
+  console.log(`Exercise 6. Circle area: ${circleArea(5)}`)
+  type TUser = {name: string, age: number}
+  const userJohn:TUser = {name: "John", age: 23}
+  const incrementAge = (user: TUser): TUser => {
+    user.age++
+    return user
+  }
+  console.log(`Exercise 6. Incremented age: ${JSON.stringify(incrementAge(userJohn))}`)
 }
 excercise6();
 
 // Create a function that takes as a first parameter an array of numbers
 // a second parameter - a function that takes a number and returns a number.
 // and returns a new array with the results of function called on each element of the array (function passed as a first parameter)
-function excercise7() {
+function excercise7(): void {
   // TODO: add type annotations
-  function map(arr, fn) {
+  type Tfun = (arg: number) => number
```

Comment:
types and interfaces name convention is ISomeName and TSomeName, where the first two chars are uppercase